### PR TITLE
Change timers process callback to physics

### PR DESCRIPTION
**Pull request by Xwdit**
Changed all timer process callbacks to physical frames so that they are framerate independent. This prevents dialogue from causing problems with certain game features that require precise timing. (e.g. replay system that can accelerate/decelerate)

This is based on a recent PR to SceneTreeTimer: https://github.com/godotengine/godot/pull/63026

### CODE REVIEW

**Review by Jowan-Spooner — APPROVED** [review]
Thanks for the PR. Looks clean and logical to me.

**Comment by coppolaemilio** [comment]
I think it makes sense to offer this as a feature, but you don't always want the timers to be based on frames, some times you need to sync with time, so an option should be added to select which one you want. 

Do you think you can make that change to the PR? 
It would require a new setting and check that setting for the timers to be set either by frame or time.

**Comment by Xwdit** [comment]
> I think it makes sense to offer this as a feature, but you don't always want the timers to be based on frames, some times you need to sync with time, so an option should be added to select which one you want.
> 
> Do you think you can make that change to the PR? It would require a new setting and check that setting for the timers to be set either by frame or time.

Even if the timer process callback is changed to physics frame, it still be sync with time in seconds (based on delta time of physics frame), So it shouldn't cause any problems to regular users.  

But I still added the corresponding setting for some special cases, or for someone doesn't like timing in physical frames. (Idle process callback is the default value)  

We may also be able to instructing users to enable physical frame timing in the documentation to solve issues caused by erratic idle timing?

File: addons/dialogic/Editor/Settings/Settings_General.gd [hunk 1]
```diff
@@ -65,3 +66,7 @@ func custom_testing_scene_selected(path:String):
 	%TestingSceneLabel.text = path
 	ProjectSettings.set_setting('dialogic/editor/test_dialog_scene', path)
 	ProjectSettings.save()
+
+func _on_physics_timer_button_toggled(button_pressed:bool):
```
Comment (Jowan-Spooner) [inline addons/dialogic/Editor/Settings/Settings_General.gd:70]: `-> void:`

File: addons/dialogic/Other/DialogicUtil.gd [hunk 2]
```diff
@@ -164,3 +164,11 @@ static func get_color_palette(default:bool = false) -> Dictionary:
 static func get_color(value:String) -> Color:
 	var colors = get_color_palette()
 	return colors[value]
+
+
+static func is_physics_timer()->bool:
+	return get_project_setting('dialogic/timer/process_in_physics', false)
+	
+
+static func update_timer_process_callback(timer:Timer):
```
Comment (Jowan-Spooner) [inline addons/dialogic/Other/DialogicUtil.gd:173]: ` -> void:`

File: addons/dialogic/Editor/VisualEditor/VisualEditor.gd [hunk 3]
```diff
@@ -175,7 +175,7 @@ func load_batch(data):
 
 func _on_batch_loaded():
 	if _batches.size() > 0:
-		await get_tree().create_timer(0.01).timeout
+		await get_tree().create_timer(0.01, true, DialogicUtil.is_physics_timer()).timeout
```
Comment (Jowan-Spooner) [inline addons/dialogic/Editor/VisualEditor/VisualEditor.gd:178]: Might not be a problem to do this here, but as this is in the editor, not in game, I'm not sure if it's necessary/good practice.

Comment (coppolaemilio) [reply addons/dialogic/Editor/VisualEditor/VisualEditor.gd:178]: Yeah, I think I will revert this one since it is in-editor

**Pull request merged**